project_task_materials_stock/models/project.py

```diff
@@ -0,0 +1,151 @@
+    def _check_stock_state(self):
+        if not self.stock_move_ids:
+            self.stock_state = 'pending'
+        elif self.stock_move_ids.filtered(lambda r: r.state == 'confirmed'):
+            self.stock_state = 'confirmed'
+        elif self.stock_move_ids.filtered(lambda r: r.state == 'assigned'):
+            self.stock_state = 'assigned'
+        elif self.stock_move_ids.filtered(lambda r: r.state == 'done'):
+            self.stock_state = 'done'
+
+    stock_move_ids = fields.One2many(
+        comodel_name='stock.move', compute='_get_stock_move',
+        string='Stock Moves')
+    analytic_line_ids = fields.One2many(
+        comodel_name='account.analytic.line', compute='_get_analytic_line',
+        string='Analytic Lines')
+    consume_material = fields.Boolean(related='stage_id.consume_material')
+    stock_state = fields.Selection(
+        [('pending', 'Pending'),
+         ('confirmed', 'Confirmed'),
+         ('assigned', 'Assigned'),
+         ('done', 'Done')], compute='_check_stock_state', string='Stock State')
+
+    @api.one
+    def unlink_stock_move(self):
+        for move in self.stock_move_ids:
+            if move.state == 'assigned':
+                move.do_unreserve()
+            if move.state in ['waiting', 'confirmed', 'assigned']:
+                move.state = 'draft'
+            move.unlink()
+
+    @api.one
+    def write(self, vals):
+        res = super(Task, self).write(vals)
+        if 'stage_id' in vals:
+            if self.consume_material:
+                self.material_ids.create_stock_move()
+                self.material_ids.create_analytic_line()
+            else:
+                self.unlink_stock_move()
+                self.analytic_line_ids.unlink()
+        return res
+
+    @api.multi
+    def action_assign(self):
+        self.stock_move_ids.action_assign()
+
+    @api.multi
+    def action_done(self):
+        self.stock_move_ids.action_done()
+        pass
+
+
+class ProjectTaskMaterials(models.Model):
+    _inherit = "project.task.materials"
+    stock_move_id = fields.Many2one(
+        comodel_name='stock.move', string='Stock Move')
+    analytic_line_id = fields.Many2one(
+        comodel_name='account.analytic.line', string='Analytic Line')
+
+    def _prepare_stock_move(self):
+        product = self.product_id
+        res = {
+            'product_id': product.id,
+            'name': product.partner_ref,
+            'state': 'confirmed',
+            'product_uom': product.uom_id.id,
+            'product_uos': product.uos_id and product.uos_id.id or False,
+            'product_uom_qty': self.quantity,
+            'product_uos_qty': self.quantity,
+            'origin': self.task_id.name,
+            'location_id':
+            product.product_tmpl_id.property_stock_procurement.id,
+            'location_dest_id': self.env.ref(
+                'stock.stock_location_customers').id,
+        }
+        if product.uos_id and product.uom_id and \
```

Comment:
hehe, I have say nothing in the previous comment. Here you do the conversion, so perfect in this way. But now I see that you should ask for which UoM to use, because maybe people want to work with another one than UoS.